File: src/web/Controllers/SECController.cs
using System.Threading.Tasks;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;
using core.fs.Portfolio;
using web.Utils;

#nullable enable

namespace web.Controllers;

[ApiController]
[Authorize]
[Route("api/[controller]")]
public class SECController(SECHandler handler) : ControllerBase
{
    [HttpGet("search")]
    public Task<ActionResult> Search([FromQuery] string query) =>
        this.OkOrError(handler.Handle(new SearchCompanies(query)));
    
    [HttpGet("filings/{ticker}")]
    public Task<ActionResult> GetFilings([FromRoute] string ticker) =>
        this.OkOrError(handler.Handle(new GetFilingsForTicker(ticker)));
    
    [HttpGet("portfolio-filings")]
    public Task<ActionResult> GetPortfolioFilings() =>
        this.OkOrError(handler.Handle(new GetPortfolioFilings(User.Identifier())));
}

File: src/web.fs/Controllers/SECController.fs
namespace web.Controllers

open System.Threading.Tasks
open Microsoft.AspNetCore.Mvc
open Microsoft.AspNetCore.Authorization
open core.fs.Portfolio
open web.Utils

[<ApiController>]
[<Authorize>]
[<Route("api/[controller]")>]
type SECController(handler: SECHandler) =
    inherit ControllerBase()

    [<HttpGet("search")>]
    member this.Search([<FromQuery>] query: string) =
        this.OkOrError(handler.Handle({SearchCompanies.Query = query}))

    [<HttpGet("filings/{ticker}")>]
    member this.GetFilings([<FromRoute>] ticker: string) =
        this.OkOrError(handler.Handle({GetFilingsForTicker.Ticker = ticker}))

    [<HttpGet("portfolio-filings")>]
    member this.GetPortfolioFilings() =
        this.OkOrError(handler.Handle({GetPortfolioFilings.UserId = this.User.Identifier()}))
